src/App.vue

```diff
@@ -325,31 +321,39 @@ export default class App extends Mixins(BcolMixin, DateMixin, FilingTemplateMixi
       // reset errors in case this method is invoked more than once (ie, retry)
       this.resetFlags()
 
-      // get business ID query param
-      const businessId = this.$route?.query?.businessId as string
-      if (!businessId) {
-        this.fetchErrorDialog = true
-        if (!businessId) throw new Error('Invalid business identifier')
-      }
-      this.setBusinessId(businessId)
-
       // ensure user is authorized or is staff to access this business
-      await this.checkAuth(businessId).catch(error => {
+      await this.checkAuth(this.getBusinessId).catch(error => {
         this.accountAuthorizationDialog = true
         throw new Error(`Auth error: ${error}`)
       })
 
-      // fetch IA filing to alter or correct
-      const { filing } = await this.fetchFiling(this.INCORPORATION_APPLICATION) // TEMP FOR TESTING
-      // const { filing } = await this.fetchFiling(this.ALTERATION) // FUTURE STATE
-
-      // parse filing into store
-      this.parseIncorpFiling(filing)
+      // check for necessary query params
```

Comment:
This code is duplicated in the views (ie, to protect themselves), but having it here (even though it's executed AFTER the views are created/mounted) means App.vue can put up the error dialog.

I don't love the duplication. Thoughts?

Reply:
Can we leave the functionality in the view and emit back to the `App` either the type or error dialog prompt?

Reply:
Done (as you suggested).